src/modules/module_20200.c

```diff
@@ -0,0 +1,331 @@
+  
+  return pw_max;
+}
+
+char *module_jit_build_options (MAYBE_UNUSED const hashconfig_t *hashconfig, MAYBE_UNUSED const user_options_t *user_options, MAYBE_UNUSED const user_options_extra_t *user_options_extra, MAYBE_UNUSED const hashes_t *hashes, MAYBE_UNUSED const hc_device_param_t *device_param)
+{
+  char *jit_build_options = NULL;
+  hc_asprintf (&jit_build_options, "-D NO_UNROLL");
+  
+  return jit_build_options;
+}
+
+int module_hash_decode (MAYBE_UNUSED const hashconfig_t *hashconfig, MAYBE_UNUSED void *digest_buf, MAYBE_UNUSED salt_t *salt, MAYBE_UNUSED void *esalt_buf, MAYBE_UNUSED void *hook_salt_buf, MAYBE_UNUSED hashinfo_t *hash_info, const char *line_buf, MAYBE_UNUSED const int line_len)
+{
+  u64 *digest = (u64 *) digest_buf;
+
+  pbkdf2_sha512_t *pbkdf2_sha512 = (pbkdf2_sha512_t *) esalt_buf;
+
+  token_t token;
+
+  token.token_cnt  = 5;
+
+  token.signatures_cnt    = 1;
+  token.signatures_buf[0] = SIGNATURE_PASSLIB_PBKDF2_SHA512;
+
+  // the hash starts with a $
+  token.sep[0]     = '$';
+  token.len_min[0] = 0;
+  token.len_max[0] = 0;
+  token.attr[0]    = TOKEN_ATTR_VERIFY_LENGTH;
+
+  token.sep[1]     = '$';
+  token.len_min[1] = 13;
+  token.len_max[1] = 13;
+  token.attr[1]    = TOKEN_ATTR_VERIFY_LENGTH
+                   | TOKEN_ATTR_VERIFY_SIGNATURE;
+
+  // iterations in decimal representation
+  token.sep[2]     = '$';
+  token.len_min[2] = 1;
+  token.len_max[2] = 6;
+  token.attr[2]    = TOKEN_ATTR_VERIFY_LENGTH
+                   | TOKEN_ATTR_VERIFY_DIGIT;
+
+  // salt in alternate base64 repretentation
+  token.sep[3]     = '$';
+  token.len_min[3] = SALT_MIN;
+  token.len_max[3] = SALT_MAX;
+  token.attr[3]    = TOKEN_ATTR_VERIFY_LENGTH
+                   | TOKEN_ATTR_VERIFY_BASE64B;
+
+  // payload in alternate base64 representanion
+  token.sep[4]     = '$';
+  token.len_min[4] = HASH_LEN_B64;
+  token.len_max[4] = HASH_LEN_B64;
+  token.attr[4]    = TOKEN_ATTR_VERIFY_LENGTH
+                   | TOKEN_ATTR_VERIFY_BASE64B;
+
+  const int rc_tokenizer = input_tokenizer ((const u8 *) line_buf, line_len, &token);
+
+  if (rc_tokenizer != PARSER_OK) return (rc_tokenizer);
+
+  // iter
+  const u8 *iter_pos = token.buf[2];
+  salt->salt_iter = hc_strtoul ((const char *) iter_pos, NULL, 10) - 1;
+
+
+  // base64 decode salt
+  const u8 *salt_pos = token.buf[3];
+  const int salt_len = token.len[3];
+  
+  u8 tmp_buf[100] = { 0 };
+  u8 tmp_buf2[100] = { 0 };
+
+  // replace . with + for proper base64 decoding
+  memcpy(tmp_buf2, salt_pos, salt_len);
+  for (int i = 0; i < salt_len; i++)
+  {
+    if (tmp_buf2[i] == '.')
+    {
+      tmp_buf2[i] = '+';
+    }
+  }
+
+  
+  const int salt_len_decoded = base64_decode (base64_to_int, (const u8 *) tmp_buf2, salt_len, tmp_buf);
+
+  u8 *salt_buf_ptr = (u8 *) pbkdf2_sha512->salt_buf;
+  memcpy (salt_buf_ptr, tmp_buf, salt_len_decoded);
+  memcpy (salt->salt_buf, salt_buf_ptr, salt_len_decoded);
+  
+  salt->salt_len    = salt_len_decoded;
```

Comment:
strange identation (spaces). I would suggest to just use salt->salt_len = salt_len_decoded; with one space before and one space after the equal sign